### ISSUE TITLE
Can't render Kitchen data set from Pixar

### ISSUE THREAD

**Issue by predat** [rodeofx/OpenWalter#4]
**Version**: openwalter 1.0.1 (from release page)
**OS**: Centos 7.4.1708
**Maya Version**: 2018.3
**MtoA Version**: 3.0.1.1

Description
---
Arnold abort when trying to render USD Kitchen Set from Pixar. 

Steps to Reproduce
---
- Create a walterStandin
- Load Kitchen_set.usd (from: http://graphics.pixar.com/usd/files/Kitchen_set.zip)
- Create an Arnold area light
- Press 'Render'

Debug Log
---
```
[mtoa.translator]  walterMtoATranslator: Maya node walterStandinShape1(walterStandin) does not have attribute aiSelfShadows to match parameter self_shadows on Arnold node walterStandinShape1(walter).
[mtoa.translator]  walterMtoATranslator: Maya node walterStandinShape1(walterStandin) does not have attribute aiOpaque to match parameter opaque on Arnold node walterStandinShape1(walter).
PreMenvPosingRefPose: poorly typed user-define -- too few parameters: varying
AiArrayConvert: unable to allocate array
anon:0x2b84a00a0690:anonymous1443559516.usda:/Kitchen_set/Arch_grp/Kitchen_1/Geom/Windows_Frame/WindowStill/Window_Frame_Arch/pCube79: varying user data is missing list array or is not an array -- disabling user-data 'PreMenvPosingRefPose'
```

**Comment by frenchdog** [comment]
Thanks for the report!

We took a quick look at found the problem. It should be fixed in the next release.

**Comment by frenchdog** [comment]
It is fixed internally and will be pushed to the opensource version soon.

In short, the Walter procedural for Arnold was supporting only displayColor attribute when it was in an array of size equal to the number of point (since we use it in this scenario only). The kitchen got displayColor attribute of size one (constant). We are now supporting this scenario.

![kitcheninwalter](https://user-images.githubusercontent.com/424502/44609345-4e5e5100-a7c5-11e8-81dd-895669b1731c.png).

**Comment by predat** [comment]
Thank you @frenchdog.

I have a relatively similar problem when I'm trying to render a simple sphere exported from [GafferHQ](https://github.com/GafferHQ/gaffer) in usda. At render time, error message look like this:
```
N: poorly typed user-define -- too few parameters: varying
```

The line relative to this error in the usd file:
```
    float3[] primvars:N (
        interpolation = "vertex"
    )
    float3[] primvars:N.timeSamples = {
        1.0000000298023224: [(-3.7855173e-8, -2.1855694e-8, -1), (0.86602545, 0, -0.4999999), (0.4330127, 0.75000006, -0.4999999), (-0.43301278, 0.75, -0.4999999), (-0.86602545, -7.5710346e-8, -0.4999999), (-0.43301263, -0.75000006, -0.4999999), (0.43301263, -0.75000006, -0.4999999), (0.8660254, 0, 0.50000006), (0.43301266, 0.75, 0.50000006), (-0.43301275, 0.75, 0.50000006), (-0.8660254, -7.5710346e-8, 0.50000006), (-0.4330126, -0.75, 0.50000006), (0.4330126, -0.75, 0.50000006), (-3.7855173e-8, -2.1855694e-8, 1)],
    }
```

Do you think that your fix will also solve this problem?

**Comment by frenchdog** [comment]
Maybe yes, maybe not :). Could you please provide a usd file to repro the problem on our side?

Thanks

**Comment by predat** [comment]
Sure
```
#usda 1.0
(
    defaultPrim = "sets"
)

def "sets"
{
}

def Mesh "sphere"
{
    float3[] extent = [(-0.86602545, -0.75000006, -1), (0.86602545, 0.75000006, 1)]
    int[] faceVertexCounts.timeSamples = {
        1.0000000298023224: [3, 3, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4, 3, 3, 3, 3, 3, 3],
    }
    int[] faceVertexIndices.timeSamples = {
        1.0000000298023224: [0, 2, 1, 0, 3, 2, 0, 4, 3, 0, 5, 4, 0, 6, 5, 0, 1, 6, 1, 2, 8, 7, 2, 3, 9, 8, 3, 4, 10, 9, 4, 5, 11, 10, 5, 6, 12, 11, 6, 1, 7, 12, 7, 8, 13, 8, 9, 13, 9, 10, 13, 10, 11, 13, 11, 12, 13, 12, 7, 13],
    }
    point3f[] points.timeSamples = {
        1.0000000298023224: [(-3.7855173e-8, -2.1855694e-8, -1), (0.86602545, 0, -0.4999999), (0.4330127, 0.75000006, -0.4999999), (-0.43301278, 0.75, -0.4999999), (-0.86602545, -7.5710346e-8, -0.4999999), (-0.43301263, -0.75000006, -0.4999999), (0.43301263, -0.75000006, -0.4999999), (0.8660254, 0, 0.50000006), (0.43301266, 0.75, 0.50000006), (-0.43301275, 0.75, 0.50000006), (-0.8660254, -7.5710346e-8, 0.50000006), (-0.4330126, -0.75, 0.50000006), (0.4330126, -0.75, 0.50000006), (-3.7855173e-8, -2.1855694e-8, 1)],
    }
    float3[] primvars:N (
        interpolation = "vertex"
    )
    float3[] primvars:N.timeSamples = {
        1.0000000298023224: [(-3.7855173e-8, -2.1855694e-8, -1), (0.86602545, 0, -0.4999999), (0.4330127, 0.75000006, -0.4999999), (-0.43301278, 0.75, -0.4999999), (-0.86602545, -7.5710346e-8, -0.4999999), (-0.43301263, -0.75000006, -0.4999999), (0.43301263, -0.75000006, -0.4999999), (0.8660254, 0, 0.50000006), (0.43301266, 0.75, 0.50000006), (-0.43301275, 0.75, 0.50000006), (-0.8660254, -7.5710346e-8, 0.50000006), (-0.4330126, -0.75, 0.50000006), (0.4330126, -0.75, 0.50000006), (-3.7855173e-8, -2.1855694e-8, 1)],
    }
    float2[] primvars:uv (
        interpolation = "faceVarying"
    )
    float2[] primvars:uv.timeSamples = {
        1.0000000298023224: [(0.083333336, 0), (0.25, 0), (0.4166667, 0), (0.5833333, 0), (0.75, 0), (0.9166666, 0), (0, 0.33333334), (0.16666667, 0.33333334), (0.33333334, 0.33333334), (0.5, 0.33333334), (0.6666667, 0.33333334), (0.8333333, 0.33333334), (1, 0.33333334), (0, 0.6666667), (0.16666667, 0.6666667), (0.33333334, 0.6666667), (0.5, 0.6666667), (0.6666667, 0.6666667), (0.8333333, 0.6666667), (1, 0.6666667), (0.083333336, 1), (0.25, 1), (0.4166667, 1), (0.5833333, 1), (0.75, 1), (0.9166666, 1)],
    }
    int[] primvars:uv:indices = [0, 7, 6, 1, 8, 7, 2, 9, 8, 3, 10, 9, 4, 11, 10, 5, 12, 11, 6, 7, 14, 13, 7, 8, 15, 14, 8, 9, 16, 15, 9, 10, 17, 16, 10, 11, 18, 17, 11, 12, 19, 18, 13, 14, 20, 14, 15, 21, 15, 16, 22, 16, 17, 23, 17, 18, 24, 18, 19, 25]
    uniform token subdivisionScheme = "none"
    matrix4d xformOp:transform.timeSamples = {
        1.0000000298023224: ( (1, 0, 0, 0), (0, 1, 0, 0), (0, 0, 1, 0), (0, 0, 0, 1) ),
    }
    uniform token[] xformOpOrder = ["xformOp:transform"]
}

```

**Comment by frenchdog** [comment]
I just give it a try and could render your low-res sphere in Arnold using our latest internal build!
It will be merged soon to Open Walter.

**Comment by rodeo** [comment]
Fixed in pull request #8
